testing/e2e/e2e_api_test.go

```diff
@@ -48,3 +57,191 @@ func (s *BeaconKitE2ESuite) TestBeaconAPIStartup() {
 	s.Require().NotEmpty(stateRootResp)
 	s.Require().False(stateRootResp.Data.IsZero())
 }
+
+// TestBeaconValidatorsWithIndices tests the beacon node api for beacon validators with indices.
+func (s *BeaconKitE2ESuite) TestBeaconValidatorsWithIndices() {
+	client := s.initBeaconTest()
+
+	indices := []phase0.ValidatorIndex{0}
+	validatorsResp, err := client.Validators(
+		s.Ctx(),
+		&beaconapi.ValidatorsOpts{
+			State:   utils.StateIDHead,
+			Indices: indices,
+		},
+	)
+	s.Require().NoError(err)
+	s.Require().NotNil(validatorsResp)
+
+	validatorData := validatorsResp.Data
+	s.Require().NotNil(validatorData, "Validator data should not be nil")
+	s.Require().
+		Len(validatorData, len(indices), "Number of validator responses should match number of requested indices")
+
+	for _, validator := range validatorData {
```

Comment:
question: if these are genesis validators, should we compare the response against them and make sure they match?
All these checks are pretty loose in my opinion and the test as is does not show completeness (is the API returning all validators that it should)?

Reply:
This specific test is only querying for 1 validator with index 0. It does not need a for loop.

Reply:
@abi87 yes, If you see test `TestValidatorsEmptyIndices` we are verifying the validators that we got from the API response and what we passed i.e 5 and it matches.

Yeah, removed the loop in above case.